docs/what-is-an-operator.md

```diff
@@ -1,2 +1,49 @@
 # What is an Operator?
 
+Operators are software extensions to Kubernetes that make use of custom resources to manage applications and their components. [Kubernetes documents](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+
+In the real world, we can define Operators: as a person who operates on some task(machine, equipment, process). Let's say it works on a given task "X". What qualities are required to peform the task X?
+
+We can classify an operator qualities into three categories:
+
+* Observe
+* Analyze
+* Act
+
+Observe the X work, analyze the state of X and finally act based on the current state of X to reach the desired state. Which will make the operators task completed, once we attain the desired state.
+
+Similar work/tasks are performed by an operators. It helps us to extend the Kubernetes API to allow custom functionality or behavior based on the needs of the consumer.
+
+# Operator Pattern 
+
+![Operator internals](images/operator-internals.png)
+
+* Observe: __Events__
+* Analyze: __Current Vs Desired State__
+* Act: __Reconciler Loop__
+
+## Components: 
+
+* Controller: Register the Custom Resource Definition and adds it to the Kubernetes API Schema, exposes a new API end-point for the CRD, added watchers for the namespace to observe, runs a process/loop i.e control loop for acting based on the desired change.
```

Comment:
As defined in the [Best practices doc](https://github.com/operator-framework/community-operators/blob/master/docs/best-practices.md#running-on-cluster)- controllers are not supposed to add their CRD to the cluster - could we call the CRD out as a separate component?